datumaro/plugins/coco_format/extractor.py

```diff
@@ -281,14 +343,21 @@ def _load_annotations(self, ann, image_info=None, parsed_annotations=None):
                     )
                 )
 
-            segmentation = ann["segmentation"]
+            segmentation = self._parse_field(ann, "segmentation", (list, dict))
             if segmentation and segmentation != [[]]:
                 rle = None
 
                 if isinstance(segmentation, list):
                     if not self._merge_instance_polygons:
                         # polygon - a single object can consist of multiple parts
                         for polygon_points in segmentation:
+                            if len(polygon_points) % 2 != 0 or len(polygon_points) < 6:
+                                raise InvalidAnnotationError(
+                                    "Polygon has invalid value count %s, "
+                                    "which is not divisible by 2. "
```

Comment:
This might not be true, since you have two conditions in the `if`. IMO, better to just have two separate `if`s.

Reply:
Ok, splitted.